Add ability for course staff to always view the results

mtyaka opened the pull request:
Course staff users can now view the results without voting.
The can also see the results for polls/surveys marked as 'private'.

If user has sufficient permissions, a 'View results' link is shown on the bottom of the component in the LMS.

mtyaka committed:
Tweaks for pep8/pylint.

mtyaka committed:
Add ability to grant view results permission to additional groups.

You can now use the XBLOCK_POLL_EXTRA_VIEW_GROUPS django setting
to allow users from additional groups to view the results.

By default only course staff has permission to view the results.

antoviaque commented:
@mtyaka The test run seem to have failed, one test is giving an error:  https://travis-ci.org/mckinseyacademy/xblock-poll/builds/51343441 - I see that you had a green checkmark on the previous PR https://github.com/mckinseyacademy/xblock-poll/pull/7 , so I assume this is due to a rebase?

mtyaka commented:
@antoviaque I don't think I every got a green build. The failures don't make much sense and the tests run fine for me locally. Will try to figure out what's wrong.

mtyaka committed:
Simplify test setup and set up travis CI.

No need to manually install xblock-sdk & dependencies anymore.
Inside a fresh virtual environment, run `sh install_test_deps.sh`,
then run the tests with `python run_tests.py`. That's it.

mtyaka committed:
Add ability for staff to always view the results.

Course staff users can now view the results without voting.
The can also see the results for polls/surveys marked as 'private'.

If user has sufficient permissions, a 'View results' link is shown
on the bottom of the component in the LMS.

mtyaka commented:
@antoviaque Fixed the build & merged.
